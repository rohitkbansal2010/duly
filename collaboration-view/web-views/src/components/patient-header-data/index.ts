export { PatientHeaderData } from './patient-header-data';
