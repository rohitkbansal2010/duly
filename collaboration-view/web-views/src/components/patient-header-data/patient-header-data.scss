@import "~@styles/themes.scss";
@import "~@styles/typography.scss";

.patientHeader {
  &Container {
    display: flex;
    align-items: center;
    min-height: 6.375rem;
  }

  &Greeting {
    @include nexa-h2-bold;
    @include nexa-extra-bold-highlight;

    margin-right: 1rem;
    color: $dupage-dark-blue;

    .patientName {
      color: $dupage-light-blue;
    }
  }

  &Avatar {
    padding: 0;
    background: $white;
    border: none;
    border-radius: 50%;
  }
}

.verticalLine {
  top: 10px;
  left: 50%;
  display: block;
  width: 0.2rem;
  height: 4.375rem;
  margin-right: 10px;
  margin-left: 10px;
  background-color: #b2becc;
}
.fontSettingContainer {
  margin-bottom: 0.8rem;
}
