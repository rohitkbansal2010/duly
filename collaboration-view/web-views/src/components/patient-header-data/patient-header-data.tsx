import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { Avatar } from '@components/avatar';
import FontSizeSetting from '@components/font-size-setting/font-size-setting';
import { PATIENT_HEADER_AVATAR_SIZE } from '@constants';
import { AvatarByRole } from '@enums';
import { usePatientViewModal } from '@hooks';
import { getPatientData } from '@redux/actions';
import { RootState } from '@redux/reducers';
import { AppDispatch } from '@redux/store';
import { getSrcAvatar } from '@utils';


import styles from './patient-header-data.scss';


export const PatientHeaderData = () => {
  const dispatch: AppDispatch = useDispatch();
  const { onShowPatientViewModal } = usePatientViewModal();
  const patientData = useSelector(({ CURRENT_APPOINTMENT }: RootState) =>
    CURRENT_APPOINTMENT.patientData);
  const patientId = useSelector(({ CURRENT_APPOINTMENT }: RootState) =>
    CURRENT_APPOINTMENT.patientId);

  const patientName = patientData?.generalInfo?.humanName?.givenNames?.join(' ').trim();
  const photoData = patientData?.photo;

  useEffect(() => {
    patientId && dispatch(getPatientData(patientId));
  }, [ dispatch, patientId ]);

  return (
    <div className={styles.patientHeaderContainer} data-testid="container">
      {patientData && (
        <>
          <h2 className={styles.patientHeaderGreeting} data-testid="welcome">
            Welcome
            {patientName ? ', ' : ''}
            <span className={styles.patientName}>{patientName}</span>
          </h2>
          <button
            className={styles.patientHeaderAvatar}
            onClick={onShowPatientViewModal}
          >
            <Avatar
              src={getSrcAvatar(photoData)}
              width={PATIENT_HEADER_AVATAR_SIZE}
              role={AvatarByRole.NON_EMPLOYEE}
              hasBorder
            />
          </button>
          <div className={styles.verticalLine} />
        </>
      )}
      <div className={styles.fontSettingContainer}>
        <FontSizeSetting/>
      </div>
    </div>
  );
};
