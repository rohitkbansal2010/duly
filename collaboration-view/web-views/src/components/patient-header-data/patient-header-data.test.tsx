import { mount, ReactWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { useSelector } from 'react-redux';

import { PatientHeaderData } from './patient-header-data';

jest.mock('react-redux', () =>
  ({
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  }));

export interface MockAvatarProps {
	width: number;
	src: string;
	role: string;
	hasBorder: boolean;
}
const mockAvatar = jest.fn();
jest.mock('@components/avatar', () =>
  ({
    Avatar: (props: MockAvatarProps) => {
      mockAvatar(props);
      return null;
    },
  }));


jest.mock('@utils', () =>
  ({ getSrcAvatar: jest.fn().mockReturnValue('mockSrc') }));

jest.mock('@constants', () =>
  ({ PATIENT_HEADER_AVATAR_SIZE: 55 }));

const mockOnShowPatientViewModal = jest.fn();
jest.mock('@hooks', () =>
  ({
    usePatientViewModal: () =>
      ({ onShowPatientViewModal: mockOnShowPatientViewModal }),
  }));

const patientMockData = {
  patientData: {
    generalInfo: {
      id: 'qwerty1',
      humanName: {
        givenNames: [
          'Ana',
          'Maria',
        ],
      },
    },
    photo: 'mockPhotoData',
  },
};

let wrapper: ReactWrapper;

beforeEach(() => {
  (useSelector as jest.Mock).mockImplementation((selector: (state: unknown) => unknown) =>
    selector({ CURRENT_APPOINTMENT : patientMockData }));

  wrapper = mount(<PatientHeaderData />);
});

describe('PatientHeaderData', () => {
  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should contain span with patientName', () => {
    expect(wrapper.containsMatchingElement(<span>Ana Maria</span>)).toEqual(true);
  });

  it('should pass photoData into getSrcAvatar', () => {
    const getSrcAvatar = require('@utils').getSrcAvatar;

    expect(getSrcAvatar).toHaveBeenCalledWith(patientMockData.patientData.photo);
  });

  it('should call button clickHandler', () => {
    wrapper.find('button').simulate('click');

    expect(mockOnShowPatientViewModal).toHaveBeenCalled();
  });

  it('should render container if there is patientData', () => {
    expect(wrapper.find({ 'data-testid': 'container' }).exists()).toBe(true);
  });

  it('should render container if there is no patientData', () => {
    (useSelector as jest.Mock).mockImplementation((selector: (state: unknown) => unknown) =>
      selector({ CURRENT_APPOINTMENT: {} }));

    const noPatientDataWrapper = mount(<PatientHeaderData />);

    expect(noPatientDataWrapper.find({ 'data-testid': 'container' }).exists()).toBe(true);
  });

  it('should render "," if there is patientName', () => {
    const text = wrapper.find({ 'data-testid': 'welcome' }).text();

    expect(/,/.test(text)).toBe(true);
  });

  it('should not render "," if there is no patientName', () => {
    patientMockData.patientData.generalInfo.humanName.givenNames = [];

    (useSelector as jest.Mock).mockImplementation((selector: (state: unknown) => unknown) =>
      selector({ CURRENT_APPOINTMENT: patientMockData }));

    const noPatientNameWrapper = mount(<PatientHeaderData />);

    const text = noPatientNameWrapper.find({ 'data-testid': 'welcome' }).text();

    expect(/,/.test(text)).toBe(false);
  });

  it('should pass correct width, src, role and hasBorder props in nested Avatar', () => {
    const mockAvatarProps: MockAvatarProps = {
      width: 55,
      src: 'mockSrc',
      role: 'NonEmployee',
      hasBorder: true,
    };
    expect(mockAvatar).toHaveBeenCalledWith(
      expect.objectContaining(mockAvatarProps)
    );
  });
});
