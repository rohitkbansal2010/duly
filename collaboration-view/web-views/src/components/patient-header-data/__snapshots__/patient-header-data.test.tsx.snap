// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientHeaderData should match snapshot 1`] = `
<PatientHeaderData>
  <div
    className="patientHeaderContainer"
    data-testid="container"
  >
    <h2
      className="patientHeaderGreeting"
      data-testid="welcome"
    >
      Welcome
      , 
      <span
        className="patientName"
      >
        Ana Maria
      </span>
    </h2>
    <button
      className="patientHeaderAvatar"
      onClick={[MockFunction]}
    >
      <Avatar
        hasBorder={true}
        role="NonEmployee"
        src="mockSrc"
        width={55}
      />
    </button>
    <div
      className="verticalLine"
    />
    <div
      className="fontSettingContainer"
    >
      <FontSizeSetting>
        <div
          className="fontSetting"
        >
          <Row>
            <div
              className="row"
            >
              <Col>
                <div
                  className="col"
                >
                  <div
                    className="fontSizes"
                  >
                    <Nav
                      defaultActiveKey="font-small"
                      fill={false}
                      justify={false}
                      onSelect={[Function]}
                    >
                      <Nav
                        activeKey="font-small"
                        as="div"
                        className="nav"
                        onSelect={[Function]}
                      >
                        <div
                          className="nav"
                          onKeyDown={[Function]}
                        >
                          <NavItem>
                            <div
                              className="nav-item"
                            >
                              <NavLink
                                disabled={false}
                                eventKey="font-small"
                              >
                                <Anchor
                                  className="nav-link active"
                                  data-rr-ui-event-key="font-small"
                                  disabled={false}
                                  onClick={[Function]}
                                >
                                  <a
                                    className="nav-link active"
                                    data-rr-ui-event-key="font-small"
                                    onClick={[Function]}
                                    onKeyDown={[Function]}
                                    role="button"
                                    tabIndex={0}
                                  >
                                    <div
                                      className="fontSizeSmall"
                                    >
                                      <img
                                        alt="font-small"
                                        src="font-icon-blue.svg"
                                      />
                                    </div>
                                  </a>
                                </Anchor>
                              </NavLink>
                            </div>
                          </NavItem>
                          <NavItem>
                            <div
                              className="nav-item"
                            >
                              <NavLink
                                disabled={false}
                                eventKey="font-medium"
                              >
                                <Anchor
                                  className="nav-link"
                                  data-rr-ui-event-key="font-medium"
                                  disabled={false}
                                  onClick={[Function]}
                                >
                                  <a
                                    className="nav-link"
                                    data-rr-ui-event-key="font-medium"
                                    onClick={[Function]}
                                    onKeyDown={[Function]}
                                    role="button"
                                    tabIndex={0}
                                  >
                                    <div
                                      className="fontSizeMedium"
                                    >
                                      <img
                                        alt="font-medium"
                                        src="font-icon-blue.svg"
                                      />
                                    </div>
                                  </a>
                                </Anchor>
                              </NavLink>
                            </div>
                          </NavItem>
                          <NavItem>
                            <div
                              className="nav-item"
                            >
                              <NavLink
                                disabled={false}
                                eventKey="font-large"
                              >
                                <Anchor
                                  className="nav-link"
                                  data-rr-ui-event-key="font-large"
                                  disabled={false}
                                  onClick={[Function]}
                                >
                                  <a
                                    className="nav-link"
                                    data-rr-ui-event-key="font-large"
                                    onClick={[Function]}
                                    onKeyDown={[Function]}
                                    role="button"
                                    tabIndex={0}
                                  >
                                    <div
                                      className="fontSizeLarge"
                                    >
                                      <img
                                        alt="font-large"
                                        src="font-icon-blue.svg"
                                      />
                                    </div>
                                  </a>
                                </Anchor>
                              </NavLink>
                            </div>
                          </NavItem>
                        </div>
                      </Nav>
                    </Nav>
                  </div>
                </div>
              </Col>
            </div>
          </Row>
          <Row>
            <div
              className="row"
            >
              <Col>
                <div
                  className="col"
                >
                  <FormRange
                    className="formRange"
                    max="3"
                    min="1"
                    onChange={[Function]}
                    step="1"
                    value="1"
                  >
                    <input
                      className="formRange form-range"
                      max="3"
                      min="1"
                      onChange={[Function]}
                      step="1"
                      type="range"
                      value="1"
                    />
                  </FormRange>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </FontSizeSetting>
    </div>
  </div>
</PatientHeaderData>
`;
